spring-boot/admin/src/main/java/eu/coatrack/admin/service/GitService.java

```diff
@@ -104,6 +107,7 @@ public void addProxy(Proxy proxy) throws GitAPIException, URISyntaxException, Fi
         PrintWriter writer = new PrintWriter(resultDir + "/ygg-proxy-" + proxy.getId() + ".yml", "UTF-8");
         writer.println("proxy-id: " + proxy.getId());
         writer.println("ygg.admin.api-base-url: " + adminApiBaseUrlForGateway);
+        writer.println("ygg.admin.server.url: " + adminServerUrl);
```

Comment:
This parameter will be missing in the config of gateways that were created before. This means that you will not be able to replace an existing gateway jar by a new version without manually changing the config.

Having this in mind, a more elegant solution could be to use the existing parameter `ygg.admin.api-base-url` and "extract the base URL" from there.

Reply:
Without `ygg.admin.api-base-url` the gateway would not be able to "talk to admin" at all, so it has to be in the config. I do not know which files you checked, they are in the individual config files per gateway in the config server's Git repo.

Reply:
Check the Anthill settings repo for the Gateway profiles